extensions/positron-r/src/provider.ts

```diff
@@ -258,198 +237,101 @@ export async function makeMetadata(
 	return metadata;
 }
 
-// directory(ies) where this OS is known to keep its R installations
-function rHeadquarters(): string[] {
-	switch (process.platform) {
-		case 'darwin':
-			return [path.join('/Library', 'Frameworks', 'R.framework', 'Versions')];
-		case 'linux':
-			return [path.join('/opt', 'R')];
-		case 'win32': {
-			const paths = [
-				path.join(process.env['ProgramW6432'] || 'C:\\Program Files', 'R')
-			];
-			if (process.env['LOCALAPPDATA']) {
-				paths.push(path.join(process.env['LOCALAPPDATA'], 'Programs', 'R'));
-			}
-			return [...new Set(paths)];
-		}
-		default:
-			throw new Error('Unsupported platform');
-	}
-}
+// functions relating to the current R binary
+let cachedRBinaryCurrent: RBinary | undefined;
```

Comment:
All of the functions that find (some definition) of the current R installation now have this caching treatment. That's because this code usually gets hit around the same time by `$validateMetadata` of the `RRuntimeManager` and by `rRuntimeDiscoverer()`. There's no point in running any of it twice, so this keeps the logs quieter and is presumably more performant in some very minor way.

If we decide to add the ability to proactively re-run R discovery (see #5223), we'll obviously clear this cached info.